apps/studio/src/lib/db/clients/sqlserver.ts

```diff
@@ -844,44 +849,28 @@ export class SQLServerClient extends BasicDatabaseClient<SQLServerResult> {
   }
 
   async importBeginCommand(_table: TableOrView, { clientExtras }: ImportFuncOptions): Promise<any> {
-    return clientExtras.transaction.begin()
+    await clientExtras.transaction.begin()
   }
 
   async importTruncateCommand (table: TableOrView, { clientExtras, executeOptions }: ImportFuncOptions): Promise<any> {
     const { name, schema } = table
     const schemaString = schema ? `${this.wrapIdentifier(schema)}.` : ''
-    return clientExtras.request.query(`TRUNCATE TABLE ${schemaString}${this.wrapIdentifier(name)};`, executeOptions)
+    await clientExtras.request.query(`TRUNCATE TABLE ${schemaString}${this.wrapIdentifier(name)};`, executeOptions)
   }
 
-  async importLineReadCommand (_table: TableOrView, sqlString: string, { clientExtras, executeOptions }: ImportFuncOptions): Promise<any> {
-    return clientExtras.request.query(sqlString, executeOptions)
+  async importLineReadCommand (table: TableOrView, sqlString: string, { executeOptions }: ImportFuncOptions): Promise<any> {
+    const { name, schema } = table
+    const schemaString = schema ? `${this.wrapIdentifier(schema)}.` : ''
+    const query = `SET IDENTITY_INSERT ${schemaString}${this.wrapIdentifier(name)} ON; ${sqlString}; SET IDENTITY_INSERT ${schemaString}${this.wrapIdentifier(name)} OFF;`;
+    return await executeOptions.request.query(query, executeOptions)
   }
```

Comment:
This is so we can actually import primary keys. Unfortunately, this apparently throws an error if the table does not have a primary key. So either we need to first check if there is a primary key on the table before importing, OR we just prevent users from mapping to primary key columns.